ant-networking/src/driver.rs

```diff
@@ -442,23 +442,25 @@ impl SwarmDriver {
 
     /// get closest k_value the peers from our local RoutingTable. Contains self.
     /// Is sorted for closeness to self.
-    pub(crate) fn get_closest_k_value_local_peers(&mut self) -> Vec<PeerId> {
+    pub(crate) fn get_closest_k_value_local_peers(&mut self) -> Vec<(PeerId, Addresses)> {
         let k_bucket_key = NetworkAddress::from_peer(self.self_peer_id).as_kbucket_key();
 
         // get closest peers from buckets, sorted by increasing distance to us
         let peers: Vec<_> = self
             .swarm
             .behaviour_mut()
             .kademlia
-            .get_closest_local_peers(&k_bucket_key)
```

Comment:
Both the fns are the same, but the other one returns the `KadPeer` which hold the addresses as well. It also ignores the passed in `self.peer_id`, but we already chain that at the end.